spaceship/lib/spaceship/connect_api/client.rb

```diff
@@ -347,8 +348,25 @@ def add_beta_groups_to_build(build_id: nil, beta_group_ids: [])
         handle_response(response)
       end
 
+      def request(method, url_or_path = nil, params = nil, headers = {}, auto_paginate = false, &block)
+        headers['Authorization'] = "Bearer #{token.text}"
+        super
+      end
+
       protected
 
+      def token
```

Comment:
This method also needs a way to pass in `key_id`, `issuer_id`, and `key`. Using environment variable won't be the only way that people will want to use this 🙃

Reply:
In this case, what do you think about changing `login` method to accept either user/pwd or key_id/key/issuer_id?

I think the user should provide this info once and the client manages refreshing the token as necessary, that's why I didn't expose the `token` method.

It seems to align with the way to programmatically use spaceship.
1. require
2. login
3. call your methods.

What do you think?